gusto_embedded/src/hooks/clientcredentials.ts

```diff
@@ -0,0 +1,201 @@
+// Code originally generated by Speakeasy (https://www.speakeasyapi.dev).
+
+import * as z from "zod";
+import { stringToBase64 } from "../lib/base64.js";
+import { HTTPClient } from "../lib/http.js";
+import { parse } from "../lib/schemas.js";
+import { PostV1PartnerManagedCompaniesSecurity$outboundSchema } from "../models/operations/postv1partnermanagedcompanies.js";
+import {
+  AfterErrorContext,
+  AfterErrorHook,
+  BeforeRequestContext,
+  BeforeRequestHook,
+  SDKInitHook,
+  SDKInitOptions,
+} from "./types.js";
+
+type Credentials = {
+  clientID: string;
+  clientSecret: string;
+};
+
+type Session = {
+  credentials: Credentials;
+  token: string;
+  expiresAt?: number;
+  scopes: string[];
+};
+
+export class ClientCredentialsHook
```

Comment:
Defines `beforeRequest` and `afterError` hooks.
* `beforeRequest` – Retrieves a [saved system access token](https://github.com/Gusto/gusto-typescript-client/pull/7/files#diff-e009aa59985ff2f1d8b84e5762fd8be3f6ee369dccd237fb33db8184665a0743R54) from an in-memory `session` **or** [creates a new system access token](https://github.com/Gusto/gusto-typescript-client/pull/7/files#diff-e009aa59985ff2f1d8b84e5762fd8be3f6ee369dccd237fb33db8184665a0743R60-R63) if there isn't a saved one. Sets this token in the authorization header.
* `afterError` – [deletes a saved token](https://github.com/Gusto/gusto-typescript-client/pull/7/files#diff-e009aa59985ff2f1d8b84e5762fd8be3f6ee369dccd237fb33db8184665a0743R91) from the session if the server returns a 401, indicating that the token is no longer valid.